huaweicloud/services/rds/data_source_huaweicloud_rds_mysql_databases_privileges.go

```diff
@@ -0,0 +1,174 @@
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	listMysqlDatabasesPrivilegesPath := listMysqlDatabasesPrivilegesClient.Endpoint + listMysqlDatabasesPrivilegesHttpUrl
+	listMysqlDatabasesPrivilegesPath = strings.ReplaceAll(listMysqlDatabasesPrivilegesPath, "{project_id}",
+		listMysqlDatabasesPrivilegesClient.ProjectID)
+	listMysqlDatabasesPrivilegesPath = strings.ReplaceAll(listMysqlDatabasesPrivilegesPath, "{instance_id}",
+		fmt.Sprintf("%v", d.Get("instance_id")))
+
+	listMysqlDatabasesPrivilegesQueryParams := buildGetMysqlDatabasesPrivilegesQueryParams(
+		fmt.Sprintf("%v", d.Get("db_name")))
+	listMysqlDatabasesPrivilegesPath += listMysqlDatabasesPrivilegesQueryParams
+
+	listMysqlDatabasesPrivilegesResp, err := pagination.ListAllItems(
+		listMysqlDatabasesPrivilegesClient,
+		"page",
+		listMysqlDatabasesPrivilegesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS Mysql databases privileges")
+	}
+
+	listMysqlDatabasesPrivilegesRespJson, err := json.Marshal(listMysqlDatabasesPrivilegesResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var listMysqlDatabasesPrivilegesRespBody interface{}
+	err = json.Unmarshal(listMysqlDatabasesPrivilegesRespJson, &listMysqlDatabasesPrivilegesRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("users", flattenListDatabasesPrivilegesBody(
+			filterListDatabasesPrivileges(d, listMysqlDatabasesPrivilegesRespBody))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetMysqlDatabasesPrivilegesQueryParams(dbName string) string {
+	return fmt.Sprintf("?db-name=%s&page=1&limit=100", dbName)
+}
+
+func flattenListDatabasesPrivilegesBody(resp []interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	rst := make([]interface{}, 0, len(resp))
+
+	for _, v := range resp {
+		rst = append(rst, map[string]interface{}{
+			"name":     utils.PathSearch("name", v, nil),
+			"readonly": utils.PathSearch("readonly", v, nil),
+		})
+	}
+	return rst
+}
+
+func filterListDatabasesPrivileges(d *schema.ResourceData, resp interface{}) []interface{} {
+	usersJson := utils.PathSearch("users", resp, make([]interface{}, 0))
+	userArray := usersJson.([]interface{})
+	if len(userArray) < 1 {
+		return nil
+	}
+	result := make([]interface{}, 0, len(userArray))
+
+	rawName, rawNameOK := d.GetOk("name")
+	rawReadonly, rawReadonlyOK := d.GetOk("readonly")
+	for _, user := range userArray {
+		name := utils.PathSearch("name", user, nil)
+		readonly := utils.PathSearch("readonly", user, nil)
+		if rawNameOK && rawName != name {
+			continue
+		}
+		if rawReadonlyOK && rawReadonly != readonly {
+			continue
+		}
```

Comment:
the type of readonly is bool, if the value is false, then this judgement logic will not take effect

Reply:
done